modules/ocf_syslog/files/ocf.conf

```diff
@@ -15,41 +25,46 @@ template(name="ocf-remote" type="list") {
     constant(value="\n")
 }
 
-# template for errored remote log lines
-template(name="ocf-errors" type="list") {
-    constant(value="ERRORED_TAG[")
-    property(name="syslogtag")
-    constant(value="]  MESSAGE[")
-    property(name="timegenerated" dateFormat="rfc3339")
-    constant(value=" ")
-    property(name="hostname")
-    constant(value=":  ")
-    property(name="msg")
-    constant(value="]\n")
+# template for remote docker log filenames
+template(name="ocf-docker-file" type="list") {
+    constant(value="/var/log/remote/services/")
+    # in jessie, syslogtag has a weird ":" appended, so we have to regex that out;
+    # the regex bits can be removed when we're totally on stretch.
+    property(name="syslogtag" regex.Expression="^[^:]+" regex.Type="ERE" regex.Match="0" securepath="replace")
+    constant(value="/")
+    property(name="syslogtag" regex.Expression="^[^:]+" regex.Type="ERE" regex.Match="0" securepath="replace")
+    constant(value=".log")
 }
 
 # template for remote log filenames
 template(name="ocf-remote-file" type="list") {
     constant(value="/var/log/remote/")
-    # in jessie, syslogtag has a weird ":" appended, so we have to regex that out;
-    # the regex bits can be removed when we're totally on stretch.
-    property(name="syslogtag" regex.Expression="^[^:]+" regex.Type="ERE" regex.Match="0" SecurePath="replace")
+    property(name="hostname" securepath="replace")
+    constant(value="/")
+    property(name="programname" caseconversion="lower" securepath="replace")
+    constant(value="/")
+    property(name="programname" caseconversion="lower" securepath="replace")
     constant(value=".log")
 }
 
 # rules for storing the logs
 ruleset(name="ocf-remote") {
-    # This regex was fiendish, since the hyphen must be next to the end bracket
-    # for it to work.
+    # The regex for the syslogtag was fiendish, since the hyphen must be next
+    # to the end bracket for it to work.
     #
     # It uses POSIX ERE (http://www.regular-expressions.info/posixbrackets.html)
     # and has a tester at http://www.rsyslog.com/regex/
     #
     # Make sure this at least matches all the items let through in the regex in
     # the stdin2syslog script so that real logs aren't put in errors.log
-    if re_match($syslogtag, '^[a-zA-Z0-9:_-]+$') then {
-        action(type="omfile" dynaFile="ocf-remote-file" template="ocf-remote" fileGroup="ocfstaff" fileCreateMode="0640")
+    #
+    # The hostname matches docker-based hostnames only (12 hex characters) to
+    # separate them out into service logs instead of per host since we go
+    # through docker hosts quite quickly and we want to group logs by service
+    # for docker, not by host.
+    if re_match($hostname, '^[a-f0-9]{12}$') and re_match($syslogtag, '^[a-zA-Z0-9:_-]+$') then {
```

Comment:
I'm not particularly happy about this, but I also don't think we're going to have hosts that have hex-based names and are 12 characters long outside of docker, so this works decently well.